pkg/sql/catalog/catprivilege/validate.go

```diff
@@ -0,0 +1,39 @@
+// Copyright 2021 The Cockroach Authors.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package catprivilege
+
+import (
+	"github.com/cockroachdb/cockroach/pkg/sql/catalog"
+	"github.com/cockroachdb/cockroach/pkg/sql/catalog/descpb"
+	"github.com/cockroachdb/cockroach/pkg/sql/privilege"
+)
+
+// Validate validates a privilege descriptor.
+func Validate(
+	p descpb.PrivilegeDescriptor, objectNameKey catalog.NameKey, objectType privilege.ObjectType,
+) error {
```

Comment:
This function and that below are convenience functions to make code more legible. The validation logic _per se_ remains in `descpb` but now requires some dependency injection with `allowedSuperuserPrivileges`.